src/main/java/flaxbeard/immersivepetroleum/common/IPSaveData.java

```diff
@@ -42,7 +43,7 @@ public IPSaveData(CompoundTag nbt){
 					ListTag islands = dim.getList("islands", Tag.TAG_COMPOUND);
 					
 					ImmersivePetroleum.log.info("[ReservoirIslands]: Read islands for dim {}", dimType.toString());
-					List<ReservoirIsland> list = islands.stream().map(inbt -> ReservoirIsland.readFromNBT((CompoundTag) inbt)).filter(o -> o != null).collect(Collectors.toList());
+					List<ReservoirIsland> list = islands.stream().map(inbt -> ReservoirIsland.readFromNBT((CompoundTag) inbt)).filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
`Nonnull` throws a NPE, which should not happen here. Anything null is being ignored/deleted.

Reply:
Oh good point, thanks